MIDAS/src/hardware/telemetry.cpp

```diff
@@ -0,0 +1,425 @@
+    // Serial.print(",");
+    // Serial.print(R"("mx":)");
+    // Serial.print(sensor_data.magnetometer_data.magnetometer.mx, 5);
+    // Serial.print(",");
+    // Serial.print(R"("my":)");
+    // Serial.print(sensor_data.magnetometer_data.magnetometer.my, 5);
+    // Serial.print(",");
+    // Serial.print(R"("mz":)");
+    // Serial.print(sensor_data.magnetometer_data.magnetometer.mz, 5);
+    // Serial.print(",");
+    // Serial.print(R"("STE_APO":)");
+    // Serial.print(sensor_data.kalman_data.kalman_apo, 5);
+    // Serial.print("");
+
+    // Serial.println("}}\n");
+}
+
+TelemetryPacket Telemetry::makePacket(RocketData Sensorstate) {
+    TelemetryPacket packet{};
+    packet.gps_lat = Sensorstate.gps.getRecent().latitude;
+    packet.gps_long = Sensorstate.gps.getRecent().longitude;
+    packet.gps_alt = Sensorstate.gps.getRecent().altitude;
+    packet.yaw = Sensorstate.orientation.getRecent().yaw;
+    packet.pitch = Sensorstate.orientation.getRecent().pitch;
+    packet.roll = Sensorstate.orientation.getRecent().roll;
+
+    // packet.gnc_state_ax = Sensorstate.kalman_data.kalman_acc_x;
+    // packet.gnc_state_vx = Sensorstate.kalman_data.kalman_vel_x;
+    // packet.gnc_state_x = Sensorstate.kalman_data.kalman_pos_x;
+    // packet.gnc_state_ay = Sensorstate.kalman_data.kalman_acc_y;
+    // packet.gnc_state_vy = Sensorstate.kalman_data.kalman_vel_y;
+    // packet.gnc_state_y = Sensorstate.kalman_data.kalman_pos_y;
+    // packet.gnc_state_az = Sensorstate.kalman_data.kalman_acc_z;
+    // packet.gnc_state_vz = Sensorstate.kalman_data.kalman_vel_z;
+    // packet.gnc_state_z = Sensorstate.kalman_data.kalman_pos_z;
+    // packet.gnc_state_apo = Sensorstate.kalman_data.kalman_apo;
+
+    //fix the gnc stuff later
+
+    packet.mag_x = inv_convert_range<int16_t>(Sensorstate.magnetometer.getRecent().mx, 8);
+    packet.mag_y = inv_convert_range<int16_t>(Sensorstate.magnetometer.getRecent().my, 8);
+    packet.mag_z = inv_convert_range<int16_t>(Sensorstate.magnetometer.getRecent().mz, 8);
+    packet.gyro_x = inv_convert_range<int16_t>(Sensorstate.orientation.getRecent().gx, 8192);
+    packet.gyro_y = inv_convert_range<int16_t>(Sensorstate.orientation.getRecent().gy, 8192);
+    packet.gyro_z = inv_convert_range<int16_t>(Sensorstate.orientation.getRecent().gz, 8192);
+
+    packet.response_ID = last_command_id;
+    
+#ifdef ENABLE_SILSIM_MODE
+    packet.rssi = 0;
+#else
+    packet.rssi = rf95.lastRssi();
+#endif
+    packet.voltage_battery = inv_convert_range<uint8_t>(Sensorstate.voltage.getRecent().voltage , 16);
+    //packet.FSM_State = (uint8_t)Sensorstate.rocketState_data.rocketStates[0];
+    packet.barometer_temp = inv_convert_range<int16_t>(Sensorstate.barometer.getRecent().temperature, 256);
+
+    TelemetryDataLite data{};
+    packet.datapoint_count = 0;
+    for (int8_t i = 0; i < 4; i++) {
+        packet.datapoints[i] = data;
+        packet.datapoint_count = i + (int8_t)1;
+    }
+    return packet;
```

Comment:
this is supposed to get queued packets, not sure why you are filling with blank packets